### DISCUSSION TITLE
CI for Windows on Arm64

### DISCUSSION THREAD

**Discussion by i-lina** [audacity/audacity#5564] [Development]
Hello Audacity team,

At Linaro, where I work, we collaborate with partners like Microsoft, Qualcomm, Arm etc to enable open source projects to run on Windows on Arm and sustain the ecosystem of open source projects. Audacity was considered an import open source project for the ecosystem. Over the past few weeks, we worked on resolving dependencies and compiler issues when compiling on a Windows on an ARM device and I have put up a PR. I was able to test it on my Windows machine. I was wondering how Audacity team does their CI and what options do we have to add a Windows Arm device to the CI.

**Comment by crsib** [comment]
We use GitHub CI exclusively ATM

**Reply by LWinterberg** [reply]
script being at https://github.com/audacity/audacity/blob/master/.github/workflows/build.yml

**Comment by i-lina** [comment]
Ah, thank you. Github just announced support for Arm processors will come in 2024.

**Reply by i-lina** [reply]
@LWinterberg / @crsib Would you have an estimate of a minimum configuration and run-time for a GitHub runner for Arm64?

**Reply by crsib** [reply]
No, not really. I can build Audacity on Raspberry Pi 4 on Linux without much issue though

**Comment by i-lina** [comment]
@LWinterberg / @crsib . Good news! Microsoft is our partner and we were able to work with them to select Audacity for GitHub Actions for ARM64 preview.
We do however need for your team to apply -
https://github.blog/changelog/2023-10-30-accelerate-your-ci-cd-with-arm-based-hosted-runners-in-github-actions/
https://resources.github.com/devops/accelerate-your-cicd-with-arm-and-gpu-runners-in-github-actions/
With a good estimation of the minutes it takes to make a build on windows for Arm64, number of builds per day and the configuration machine, you could apply for GitHub runners using the above link and that should be enough for now.
If we are reasonable, I am confident that we get that even sponsored for Audacity.
Message me separately, if you run into a snag and I can put you in touch with our contact in Microsoft and we can resolve this for you.

**Reply by LWinterberg** [reply]
I've filled out the form!

**Reply by i-lina** [reply]
I was informed that you should have access to GitHub Actions for ARM64 now. Could you try and let me know?

**Comment by lexcyn** [comment]
Any update on this? Would love to use a native ARM version (or be able to compile one from source)

**Reply by i-lina** [reply]
You should be able to compile from the mainline for Windows on Arm. However CI support is blocked on lack of GHA availability for Opensource projects (ETA end of 2024).

**Reply by tresf** [reply]
It's 2025.  Has there been any progress in rolling Windows ARM64 runners out to open source projects?

**Reply by i-lina** [reply]
Unfortunately no. The GHA is still unavailable for Windows for open source projects.

**Comment by i-lina** [comment]
GHA is now available for open source projects on Windows on Arm:
https://newsroom.arm.com/blog/windows-arm64-runners-git-hub-actions

Could we add Audacity CI/CD for WoA platform please?

**Reply by tresf** [reply]
Thank you!  I just toggled this on for another project (LMMS) and builds are succeeding!   Sharing the PR here incase the build strategy helps others. https://github.com/LMMS/lmms/pull/7848
